### PR TITLE
Initial implementation of the toggling features process

### PR DESCRIPTION
**Pull request by t13ka**
Now we have a features agent which reading file wwwroot/cms-content/settings_data.json
This agent reads the file and evaluates the results which can be rendered on liquid pages

### CODE REVIEW

File: VirtoCommerce.Storefront.Tests/Features/Samples/full_data.json [hunk 1]
```diff
@@ -0,0 +1,156 @@
+{
+  "features": [
+    {
+      "Name": "OrdersBrowsing",
+      "IsActive": true,
+      "Requires": [],
+      "Replaces": [],
+      "Replaced": [],
```
Comment (trueboroda) [inline VirtoCommerce.Storefront.Tests/Features/Samples/full_data.json:8]: Why are these two parameters necessary? Is this not redundant? Two sources of truth.

Comment (t13ka) [reply VirtoCommerce.Storefront.Tests/Features/Samples/full_data.json]: Agree. The replaced property was removed

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+namespace VirtoCommerce.LiquidThemeEngine.Filters
+{
+    using System;
```
Comment (tatarincev) [inline VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs:3]: Could you do not use this nested `using` declaration? This syntax looks to be inconsistent with our general code style convention. We have negotiated regarding this style in one of dev meetings and have come to an agreement to do not use nested `using` statements.

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    public class FeatureFilter
+    {
+        public static string IsFeatureActive(TemplateContext context, string key, params object[] variables)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                return IsFeaturesActive(context, key, variables);
+            }
+
+            string result;
```
Comment (tatarincev) [inline VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs:18]: this condition may be simplified by default initialization
```C#
var result = Convert.ToString(false);
if (context.TemplateLoader is ShopifyLiquidThemeEngine themeEngine)
{
...
}
``

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public static string IsFeaturesActive(TemplateContext context, string key, params object[] variables)
+        {
+            if (!string.IsNullOrEmpty(key))
+            {
+                return IsFeatureActive(context, key, variables);
+            }
+
+            string result;
```
Comment (tatarincev) [inline VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs:38]: see above

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+{
+    using System;
+
+    using global::Scriban;
+
+    using Newtonsoft.Json.Linq;
+
+    public class FeatureFilter
+    {
+        public static string IsFeatureActive(TemplateContext context, string key, params object[] variables)
```
Comment (tatarincev) [inline VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs:11]: You can return any Built-in .NET types from Liquid filters, you do not need to return only string then.

File: VirtoCommerce.LiquidThemeEngine/ShopifyLiquidThemeEngine.cs [hunk 6]
```diff
@@ -499,5 +502,24 @@ private string GetSettingsFilePath()
             var prefix = _httpContextAccessor.HttpContext.Request.Query["preview_mode"];
             return prefix.ToString().IsNullOrEmpty() ? "settings_data.json" : $"drafts\\{prefix}_settings_data.json";
         }
+
+        public bool IsFeatureActive(string featureName)
+        {
+            return _memoryCache.GetOrCreateExclusive(featureName, cacheEntry =>
```
Comment (tatarincev) [inline VirtoCommerce.LiquidThemeEngine/ShopifyLiquidThemeEngine.cs:508]: Better create cache key by this statement. This is guarantee its uniqueness
```
 var cacheKey = CacheKey.With(GetType(), nameof(IsFeatureActive), featureName);
```

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs (hunk 2 above)
Comment (t13ka) [reply VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs]: done

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs (hunk 3 above)
Comment (t13ka) [reply VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs]: I simplified the entire code

File: VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs (hunk 5 above)
Comment (t13ka) [reply VirtoCommerce.LiquidThemeEngine/Filters/FeatureFilter.cs]: see above

File: VirtoCommerce.LiquidThemeEngine/ShopifyLiquidThemeEngine.cs (hunk 6 above)
Comment (t13ka) [reply VirtoCommerce.LiquidThemeEngine/ShopifyLiquidThemeEngine.cs]: done

**Pull request merged**